desc/compute/_core.py

```diff
@@ -838,18 +848,43 @@ def compute_geometry(
             )
         )
     if check_derivs("V_r(r)", R_transform, Z_transform):
+        P = compress(  # perimeter
+            grid,
+            surface_integrals(
+                grid, jnp.sqrt(data["g_tt"]), surface_label="zeta", max_surface=True
+            ),
+            surface_label="zeta",
+        )
+        A = compress(  # area
+            grid,
+            surface_integrals(
+                grid, jnp.abs(data["sqrt(g)"] / data["R"]), surface_label="zeta"
+            ),
+            surface_label="zeta",
+        )
+        # derived from Ramanujan approximation for the perimeter of an ellipse
+        a = (  # semi-major radius
+            jnp.sqrt(3)
+            * (
+                jnp.sqrt(8 * jnp.pi * A + P**2)
+                + jnp.sqrt(
+                    2 * jnp.sqrt(3) * P * jnp.sqrt(8 * jnp.pi * A + P**2)
+                    - 40 * jnp.pi * A
+                    + 4 * P**2
+                )
+            )
+            + 3 * P
+        ) / (12 * jnp.pi)
+        b = A / (jnp.pi * a)  # semi-minor radius
         # eq. 4.9.10 in W.D. D'haeseleer et al. (1991) doi:10.1007/978-3-642-75595-8.
         data["V_r(r)"] = surface_integrals(grid, jnp.abs(data["sqrt(g)"]))
         data["S(r)"] = surface_integrals(grid, data["|e_theta x e_zeta|"])
         data["V"] = jnp.sum(jnp.abs(data["sqrt(g)"]) * grid.weights)
-        data["A"] = jnp.mean(
-            surface_integrals(
-                grid, jnp.abs(data["sqrt(g)"] / data["R"]), surface_label="zeta"
-            )
-        )
+        data["A"] = jnp.mean(A)
         data["R0"] = data["V"] / (2 * jnp.pi * data["A"])
         data["a"] = jnp.sqrt(data["A"] / jnp.pi)
         data["R0/a"] = data["V"] / (2 * jnp.sqrt(jnp.pi * data["A"] ** 3))
+        data["a/b"] = jnp.max(a / b)
```

Comment:
might be a bit confusing since the "a" here is different than data["a"]? Could call it "kappa" (common term for elongation in tokamaks) or just elongation?

Also do we want to make it mean instead of max? so it agrees more with the way we compute area?

Reply:
Maximum elongation is what we care about penalizing in optimization. If we used mean instead, we could end up with a circular cross-section (elongation=0) that "cancels out" another cross-section with very high elongation, and that would be bad.

Reply:
Regarding the name: I'm not sure what is best. I've seen "epsilon" in the stellarator literature, but that is also used for a lot of other things. I wanted to keep something similar to the other names we use like "R0/a" for aspect ratio, etc. But if you think "a" is too confusing I would change it to "elongation".

Reply:
changed to `"a_major/a_minor"`